### DISCUSSION TITLE
Get all dependencies‘ source files

### DISCUSSION THREAD

**Discussion by basbebe** [ipetkov/crane#718] [Q&A] [answered]
Is there an elegant way to access the source files of all cargo dependencies?

There is `$CARGO_HOME` if I use `cargoArtifacts` but I guess there is currently no way to access them individually using nix?

In my case I'm looking for a way to include e.g. all `LICENSE` files of all dependencies.

**Comment by ipetkov** [comment, marked as answer]
Hi @basbebe ! Technically it is possible to crawl the entire contents of the vendor directories and access crate sources that way, but I'm afraid there's no _elegant_ ways that I know of (even outside of the Nix sandbox).

If your dependency licenses are all you are interested in, I'd recommend taking a look at something like [cargo deny](https://github.com/EmbarkStudios/cargo-deny) or [cargo license](https://github.com/onur/cargo-license). Hope this helps!

**Reply by basbebe** [reply]
I am already using `cargo-deny` and `cargo-license`.

Where crates use a `license-file` or a license that needs a copyright notice to be included I would like to include those files from the source and automate this process.

So yes, crawling `$CARGO_HOME` is probably what I will have to do then.

I was hoping there was something like a `passthru` with a list of the dependencies' source derivations that I just overlooked 😅

Thanks though!
